metrics.go

```diff
@@ -237,6 +237,10 @@ func (m *metrics) countVariants(name string, enabled bool, variantName string) {
 
 	m.add(name, enabled, 1)
 	m.metricsChannels.count <- metric{Name: name, Enabled: enabled}
+
+	m.bucketMu.Lock()
+	defer m.bucketMu.Unlock()
+
```

Comment:
How does this actually work? I'm used to having the unlock after the guarded portion of the code, but maybe it's the defer what makes the difference...